### PR TITLE
[Driver] NFC: Add some "const"s to clarify the code.

### PR DESCRIPTION
**Pull request by davidungar**
All this PR does is add some "const"s to existing declarations. It is a tiny first step to improving the malleability of the driver.

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Review by jrose-apple — APPROVED** [review]

**Comment by davidungar** [comment]
@swift-ci smoke test and merge

File: include/swift/Driver/Compilation.h [hunk 1]
```diff
@@ -172,22 +172,22 @@ class Compilation {
...
-  bool EnableBatchMode;
+  const bool EnableBatchMode;
 
   /// Provides a randomization seed to batch-mode partitioning, for debugging.
-  unsigned BatchSeed;
+  const unsigned BatchSeed;
 
   /// In order to test repartitioning, set to true if
   /// -driver-force-one-batch-repartition is present. This is cleared after the
   /// forced repartition happens.
```
[the remark this answers is not in the document]
Comment (davidungar) [reply include/swift/Driver/Compilation.h]: Thanks--great catch! I guess this const'ing already found one bogus comment.

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Pull request merged**